### ISSUE TITLE
Multiple client on same project to post report on different Sentry projects

### ISSUE THREAD

**Issue by xamix** [getsentry/sentry-native#612]
Hello,

I want to know if it is possible to send report to different endpoint on Sentry.
I have a C++ project which already have a Native instance which send report to Project let's say A.

This project communicate with a remote device call it project B which send error and information to project A.
Project A need to format and push report to Sentry project B.

Is this possible and if yes, how to configure multiple DSN?
Regards

**Comment by flub** [comment]
I'm afraid this isn't possible, the canonical answer would be to let the device report the errors directly to sentry but given your question I assume this device is out of your control or otherwise can not do that.

A simple work around could be to use a tag to identify such errors, it would allow you to include it in search terms for the issue list.

How does the remote report the error btw?  If it sends a complete event payload you could make the http requests to the sentry endpoint manually maybe?

**Comment by xamix** [comment]
The remote device is an long life embedded battery system (It's programmed in C code).
We retrieve very light information via some emergency messages which send for example the assert program counter address and Git hash code.
The idea was to create a Sentry project for this embedded system, upload program and symbol on it in order to do some symbolication when we push an exception generated by interpreting those light informations from embedded system

**Comment by realkosty** [comment]
Hi @supervacuus, 
I have a similar but simpler request from another customer: 
Is it possible to send the same error to multiple Sentry projects? (no Project B involved) 

In other SDKs you just create multiple Client objects with different DSNs (e.g. sentry-python: https://github.com/getsentry/sentry-python/issues/198)

**Comment by supervacuus** [comment]
Hi, @realkosty! If I understand correctly, you ask whether sending every envelope (or only crashes) to two (or more) endpoints is possible instead of one. 

That is currently not possible. We do not support multiple "clients", and the Native SDK also doesn't implement the concept of a "hub". There is a single "scope" in the Native SDK, which means that even if we can provide multiple transport endpoints at some point, the envelopes sent to these endpoints would still have shared meta-data (which might be sufficient for your use case).

Another aspect concerning crashes is that if you use `crashpad` as a backend, we'd also have to integrate multiple endpoints to the `crashpad_handler`.

In summary: if this is what you aim for, it would be different enough from the original issue (which asks for the ability to send very different payloads to the endpoints) to justify creating a new enhancement request. This is especially true because introducing such a feature would duplicate the traffic for essentially the same data. Maybe it makes more sense to find a solution where we duplicate the events in the backend then?

CC: @kahest

**Comment by ashwoods** [comment]
@realkosty Is there a specific reason why this would need to be done client side? Using relay or a relay-like service service would be one way of achieving this.
